pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1358,13 +1379,20 @@ func (r *ReconcileClusterDeployment) syncDeletedClusterDeployment(cd *hivev1.Clu
 		return reconcile.Result{}, err
 	}
 
+	customizationReleased, err := r.ensureClusterDeploymentCustomizationIsReleased(cd, cdLog)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
 	switch {
 	case !deprovisioned:
 		return reconcile.Result{}, nil
 	case !dnsZoneGone:
 		return reconcile.Result{RequeueAfter: defaultRequeueTime}, nil
+	case !customizationReleased:
+		return reconcile.Result{}, nil
 	default:
-		cdLog.Infof("DNSZone gone and deprovision request completed, removing finalizer")
+		cdLog.Infof("DNSZone gone, customization released and deprovision request completed, removing finalizer")
```

Comment:
Customization will only be released if it exists so this does not sound right to me